### PR TITLE
Fix terrible script performance when rendering load is too high

### PR DESCRIPTION
**Pull request by jherico**
* Remove excessive blocking calls from Overlays where they're not necessary
* Modify the triggering logic for idle / paint in Application so that we wait for the next present before starting on a new frame
* Add more efficient overlay property fetching mechanisms

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1479/HighFidelity-Beta-PR10925-b1d3c3bf748293a2174bf63982d0ca7b18809ca0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1479/HighFidelity-Beta-PR10925-b1d3c3bf748293a2174bf63982d0ca7b18809ca0.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1482/HighFidelity-Beta-PR10925-cd11f5cfa6edda943b9015e13d0bc645a6f83024.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1482/HighFidelity-Beta-PR10925-cd11f5cfa6edda943b9015e13d0bc645a6f83024.exe

File: interface/src/ui/overlays/Overlays.cpp [hunk 1]
```diff
@@ -955,10 +974,11 @@ bool Overlays::mouseMoveEvent(QMouseEvent* event) {
...
     QVector<QUuid> result;
-    if (QThread::currentThread() != thread()) {
-        BLOCKING_INVOKE_METHOD(this, "findOverlays", Q_RETURN_ARG(QVector<QUuid>, result), Q_ARG(glm::vec3, center), Q_ARG(float, radius));
-        return result;
-    }
+    //if (QThread::currentThread() != thread()) {
+    //    PROFILE_RANGE(script, __FUNCTION__);
+    //    BLOCKING_INVOKE_METHOD(this, "findOverlays", Q_RETURN_ARG(QVector<QUuid>, result), Q_ARG(glm::vec3, center), Q_ARG(float, radius));
+    //    return result;
+    //}
```
Comment (sethalves) [inline interface/src/ui/overlays/Overlays.cpp]: should remove the commented code, but we can get it in a later PR

**Comment by sethalves** [comment]
to test:
- no functional change, except hopefully better tablet and grabbing and teleport performance

**Comment by CainFoool** [comment]
Testing.

**Comment by CainFoool** [comment]
# Test Passed
Test | Result
-------- | --------
no functional change, except hopefully better tablet and grabbing and teleport performance | Pass

Everything seemes to be still working okay - if anything, felt like the tablet was a bit smoother. Didn't notice much of a difference (if any) between grabbing and teleporting.

Let me know if there's anything else to do here.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

**Comment by YuppyPlays** [comment]
Testing [as per qatesters chat]

**Comment by YuppyPlays** [comment]
Testing complete:

no functional change, except hopefully better tablet and grabbing and teleport performance
**Pass, everything works the same, I'm not sure if it's just me but I found the scrolling in the Create Properties now to be smoother and less disruptive (in previous PR's/Master it seemed a tad jumpy when trying to scroll with the stylus)**

**Pull request merged**